src/main/java/at/hannibal2/skyhanni/features/mining/FlowstateHelper.kt

```diff
@@ -0,0 +1,173 @@
+import at.hannibal2.skyhanni.events.LorenzTickEvent
+import at.hannibal2.skyhanni.events.mining.OreMinedEvent
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.RegexUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.RenderUtils.renderRenderables
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.SkyBlockItemModifierUtils.getEnchantments
+import at.hannibal2.skyhanni.utils.inPartialSeconds
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration.Companion.seconds
+
+@SkyHanniModule
+object FlowstateHelper {
+    private val config get() = SkyHanniMod.feature.mining
+    private var streakEndTimer = SimpleTimeMark.farPast()
+    private var blockBreakStreak = 0
+
+    private var display: List<Renderable>? = null
+    private var displayDirty = false
+
+    private var flowstateCache: Int? = null
+    private var hasChangedItem = true
+
+    private val pickobulusPattern by RepoPattern.pattern(
+        "mining.pickobulus.blockdestroy",
+        "§7Your §r§aPickobulus §r§7destroyed §r§e(?<amount>\\d+) §r§7blocks!"
+    )
+
+    enum class FlowstateElements(val label: String, var renderable: Renderable) {
+        TITLE("§d§lFlowstate Helper", Renderable.string("§d§lFlowstate Helper")),
+        TIMER("Time Remaining: §b6.71", Renderable.string("")),
+        STREAK("Streak: §7234", Renderable.string("")),
+        SPEED("§6+600⸕", Renderable.string(""));
+
+        override fun toString() = label
+
+        companion object {
+            @JvmField
+            val defaultOption = listOf(
+                TITLE, TIMER, STREAK, SPEED
+            )
+        }
+    }
+
+    @HandleEvent(onlyOnSkyblock = true)
+    fun onBlockMined(event: OreMinedEvent) {
+        if (!MiningAPI.inCustomMiningIsland()) return
+        if (hasFlowstate() == null) return
+
+        streakEndTimer = SimpleTimeMark.now().plus(10.seconds)
+        blockBreakStreak += event.extraBlocks.values.sum()
+        createDisplay()
+        displayDirty = true
+        attemptClearDisplay()
+    }
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) { //TODO: Remove once #2540 is merged
+        if (!MiningAPI.inCustomMiningIsland()) return
+        if (hasFlowstate() == null) return
+
+        pickobulusPattern.matchMatcher(event.message) {
+            streakEndTimer = SimpleTimeMark.now().plus(10.seconds)
+            blockBreakStreak += group("amount").toInt()
+            createDisplay()
+            attemptClearDisplay()
+            displayDirty = true
+        }
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!MiningAPI.inCustomMiningIsland()) return
+
+        attemptClearDisplay()
+    }
```

Comment:
does this need to be here now that there is a world swap event?

Reply:
well, the streak is supposed to expire after 10s without mining blocks, i don't think having only the world swap would be useful

Reply:
> well, the streak is supposed to expire after 10s without mining blocks, i don't think having only the world swap would be useful

oh yeah that makes sense